MAINTENANCE.md

```diff
@@ -0,0 +1,90 @@
+
+With no further ado, the steps to release a new version are:
+
+### fury-distribution
+
+> [!WARNING]
+> If you are releasing a new `x.y.0` version create a `release-vX.<y-1>` branch for the previous release.
+
+1. Create a new branch `feat/vx.y.z` (`v1.29.4`, for example) where to work on.
+2. Create the PRs fixing the issues or adding new features to the templates or other files of fury-distribution, test them and merge them.
+3. Update the `kfd.yaml` and `Furyfile.yaml` files, bumping the distribution version, adjusting the modules and installers versions where needed.
+4. If the distribution schemas have been changed:
+   1. If you haven't already, install the needed tools with `make tools-go`.
+   2. Generate the new docs with `make generate-docs`.
+   3. Generate the go models with `make generate-go-models`
+5. Update the CI and e2e tests to point to the new version:
+   1. `.drone.yaml`
+   2. `tests/e2e-kfddistribution-*.yaml`
+   3. `tests/e2e-kfddistribution-upgrades.sh`
+   4. `tests/e2e/kfddistribution-upgrades/furyctl-init-cluster-1.29.3.yaml`
+6. Update the documentation:
+   1. `README.md`
+   2. `docs/COMPATIBILITY_MATRIX.md`
+   3. `docs/VERSIONING.md`
+   4. Write the release notes for the new version (`docs/releases/vx.y.z.md`)
+7. Tag a release candidate to trigger all the e2e tests and fix eventual problems
+
+At this point, you'll need to switch to pushing some changes in furyctl
+
+### furyctl
+
+8. Create a new branch for the WIP release like `feat/vx.y.z` (`v0.29.8`, for example)
+9. Add the new versions to the `internal/distribution/compatibility.go` file.
+10. Add the migration paths to the corresponding kinds in `configs/upgrades/{onpremises,kfddistribution,ekscluster}/`, creating the needed folders for each new version.
+11. Update the documentation:
+    1. `README.md`.
+    2. `docs/COMPATIBLITY_MATRIX.md`.
+12. Update the compatibility unit tests with the new versions (`internal/distribution/compatibility_test.go`)
+13. Bump the version to the new `fury-distribution` go library that has been released as RC in step `7`.
+
+  ```bash
+  go get -u github.com/sighupio/fury-distribution@v1.29.3
+  go mod tidy
+  ```
+
+14. Tag a release candidate with the changes. This will be used in the e2e tests of the distribution.
+
+### Back to fury-distribution
+
+14. Update the CI's `.drone.yaml` file to use the release candidate for furyctl that you released in step `14`.
+15. Update the e2e tests with the new upgrade paths.
+16. Tag a new release candidate of the distribution to run the e2e tests using the new upgrade paths and furyctl's RC.
+17. After the CI passes and the PR has been approved, merge into `main`
+18. Tag the final release and let the CI run again and do the release.
+19. **Repeat all the process for the other 2 "minor" versions that need to be updated**, but targeting `release-vx.y` branches instead of `main`.
```

Comment:
Fixed list :)

```suggestion
15. Update the CI's `.drone.yaml` file to use the release candidate for furyctl that you released in step `14`.
16. Update the e2e tests with the new upgrade paths.
17. Tag a new release candidate of the distribution to run the e2e tests using the new upgrade paths and furyctl's RC.
18. After the CI passes and the PR has been approved, merge into `main`
19. Tag the final release and let the CI run again and do the release.
20. **Repeat all the process for the other 2 "minor" versions that need to be updated**, but targeting `release-vx.y` branches instead of `main`.
```